keps/sig-api-machinery/20190726-component-config-fields-should-be-optional.md

```diff
@@ -0,0 +1,197 @@
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+`ComponentConfig` allows for specification of options for a component in a
+structured and versioned way; kubernetes apimachinery objects replace flags.
+
+However, while flags make the distinction between not-specified and
+set-to-empty-value, our api types generally do not.  By using pointer-types it
+is possible to preserve this difference, this KEP proposes that we should use
+pointer-types in our componentconfig.
+
+This also allows for generation of a minimal componentconfig from code, and
+should faciliate patching and merging of componentconfig.
+
+ComponentConfig types are used with v1alpha1 versions for kube-scheduler,
+kube-proxy, kube-controller-manager, cloud-controller-manager; I intend to make
+these nilable.  kubelet uses a ComponentConfig type that is v1beta1, I propose
+not changing those at the current time, but rather proving out the approach with
+the v1alpha1 types.
```

Comment:
Might want to put details here wrt if successful we will... This should also serve as a blocker towards graduating other component configs.